From 87723a65b350f6d14c14c740844511d3ef08c38f Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Wed, 14 Jun 2017 19:26:51 +0000
Subject: [PATCH] OS-6188 Missing build dependencies when generating message
 catalogs Reviewed by: Dan McDonald <danmcd@joyent.com> Reviewed by: Jerry
 Jelinek <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 usr/src/cmd/fs.d/nfs/mount/Makefile    | 4 ++--
 usr/src/cmd/fs.d/udfs/fsdb/Makefile    | 4 +++-
 usr/src/cmd/fs.d/udfs/labelit/Makefile | 4 +++-
 usr/src/cmd/iconv/Makefile             | 2 ++
 usr/src/cmd/isns/isnsd/Makefile        | 3 +++
 usr/src/cmd/msgfmt/Makefile            | 4 ++++
 usr/src/cmd/sgs/Makefile               | 4 +++-
 usr/src/cmd/svc/svccfg/Makefile        | 3 +++
 8 files changed, 23 insertions(+), 5 deletions(-)

diff --git a/usr/src/cmd/fs.d/nfs/mount/Makefile b/usr/src/cmd/fs.d/nfs/mount/Makefile
index c2485208a8..e9e55aac78 100644
--- a/usr/src/cmd/fs.d/nfs/mount/Makefile
+++ b/usr/src/cmd/fs.d/nfs/mount/Makefile
@@ -19,7 +19,7 @@
 # CDDL HEADER END
 #
 # Copyright (c) 1990, 2010, Oracle and/or its affiliates. All rights reserved.
-# Copyright 2014, Joyent, Inc. All rights reserved.
+# Copyright 2017, Joyent, Inc. All rights reserved.
 #
 # cmd/fs.d/nfs/mount/Makefile
 
@@ -101,7 +101,7 @@ webnfs.x:	../lib/webnfs.x
 #
 catalog: $(POFILE)
 
-$(POFILE): $(SRCS)
+$(POFILE): $(SRCS) webnfs.h
 	$(RM) $@
 	$(COMPILE.cpp) $(SRCS)   > $(POFILE).i
 	$(XGETTEXT)     $(XGETFLAGS) $(POFILE).i
diff --git a/usr/src/cmd/fs.d/udfs/fsdb/Makefile b/usr/src/cmd/fs.d/udfs/fsdb/Makefile
index bda23962f3..3c8ea7cc87 100644
--- a/usr/src/cmd/fs.d/udfs/fsdb/Makefile
+++ b/usr/src/cmd/fs.d/udfs/fsdb/Makefile
@@ -22,6 +22,8 @@
 # Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 #
+# Copyright 2017, Joyent Inc.
+#
 
 FSTYPE=		udfs
 LIBPROG=	fsdb
@@ -100,7 +102,7 @@ catalog:        $(POFILE)
 
 CATSRCS=	$(SRCS) lex.yy.c y.tab.c
 
-$(POFILE):      $(CATSRCS)
+$(POFILE):      $(CATSRCS) ud_lib.h
 	$(RM) $@
 	$(COMPILE.cpp) $(CATSRCS)   > $(POFILE).i
 	$(XGETTEXT) $(XGETFLAGS)        $(POFILE).i
diff --git a/usr/src/cmd/fs.d/udfs/labelit/Makefile b/usr/src/cmd/fs.d/udfs/labelit/Makefile
index e0bfc31b25..24a0592f0f 100644
--- a/usr/src/cmd/fs.d/udfs/labelit/Makefile
+++ b/usr/src/cmd/fs.d/udfs/labelit/Makefile
@@ -22,6 +22,8 @@
 # Copyright 2006 Sun Microsystems, Inc.  All rights reserved.
 # Use is subject to license terms.
 #
+# Copyright 2017 Joyent, Inc.
+#
 
 FSTYPE=		udfs
 LIBPROG=	labelit
@@ -63,7 +65,7 @@ POFILE= labelit.po
 #
 catalog:	$(POFILE)
 
-$(POFILE):	$(SRCS)
+$(POFILE):	$(SRCS) ud_lib.h
 	$(RM) $@
 	$(COMPILE.cpp) $(SRCS) > $(POFILE).i
 	$(XGETTEXT) $(XGETFLAGS) $(POFILE).i
diff --git a/usr/src/cmd/iconv/Makefile b/usr/src/cmd/iconv/Makefile
index 9e4a83cc18..02edd5238d 100644
--- a/usr/src/cmd/iconv/Makefile
+++ b/usr/src/cmd/iconv/Makefile
@@ -11,6 +11,7 @@
 
 #
 # Copyright 2011 Nexenta Systems, Inc.  All rights reserved.
+# Copyright 2017 Joyent Inc.
 #
 
 PROG=iconv
@@ -44,6 +45,7 @@ $(PROG): $(OBJS)
 	$(POST_PROCESS)
 
 $(OBJS):	parser.tab.h
+$(PIFILES):	parser.tab.h
 
 parser.tab.c parser.tab.h: parser.y
 	$(YACC) $(YFLAGS) parser.y
diff --git a/usr/src/cmd/isns/isnsd/Makefile b/usr/src/cmd/isns/isnsd/Makefile
index 9de0a42d7a..f104960808 100644
--- a/usr/src/cmd/isns/isnsd/Makefile
+++ b/usr/src/cmd/isns/isnsd/Makefile
@@ -25,6 +25,7 @@
 
 #
 # Copyright (c) 2012 by Delphix. All rights reserved.
+# Copyright 2017 Joyent, Inc.
 #
 
 PROG = isns
@@ -94,6 +95,8 @@ clean:
 
 lint:	lint_SRCS
 
+$(POFILES): $(DTRACE_HEADER)
+
 $(POFILE): $(POFILES)
 	$(RM) $@
 	$(CAT) $(POFILES) > $@
diff --git a/usr/src/cmd/msgfmt/Makefile b/usr/src/cmd/msgfmt/Makefile
index 63b22a8d50..4e4a626413 100644
--- a/usr/src/cmd/msgfmt/Makefile
+++ b/usr/src/cmd/msgfmt/Makefile
@@ -25,6 +25,8 @@
 #       Copyright (c) 1984, 1986, 1987, 1988, 1989 AT&T
 #         All Rights Reserved
 #
+# Copyright 2017 Joyent, Inc.
+#
 
 # cmd/msgfmt/Makefile
 
@@ -90,6 +92,8 @@ xgettext: $(XOBJS) $(LXOBJS)
 	$(LINK.c) $(XOBJS) $(LXOBJS) -o $@ $(LDLIBS)
 	$(POST_PROCESS)
 
+$(POFILES): y.tab.h
+
 $(POFILE):	$(POFILES)
 	$(RM) $@
 	cat $(POFILES) > $@
diff --git a/usr/src/cmd/sgs/Makefile b/usr/src/cmd/sgs/Makefile
index 6dfa9f91d9..f43ea16e2c 100644
--- a/usr/src/cmd/sgs/Makefile
+++ b/usr/src/cmd/sgs/Makefile
@@ -21,6 +21,7 @@
 #
 # Copyright (c) 1991, 2010, Oracle and/or its affiliates. All rights reserved.
 # Copyright 2016 RackTop Systems.
+# Copyright 2017 Joyent, Inc.
 #
 
 include		$(SRC)/cmd/Makefile.cmd
@@ -123,7 +124,8 @@ _msg: _msg_gettext _msg_sgsmsg
 
 _msg_gettext: $(MSGDOMAIN)/$(POFILE)
 
-_msg_sgsmsg: $(MSGDIR)
+# $(MACH)/sgsmsg must be built before we can descend into $(MSGDIR)
+_msg_sgsmsg: native-add .WAIT $(MSGDIR)
 
 $(MSGDOMAIN)/$(POFILE): \
 		$(MSGDOMAIN) $(POFILE)
diff --git a/usr/src/cmd/svc/svccfg/Makefile b/usr/src/cmd/svc/svccfg/Makefile
index d5a6e434d6..8fa86739a0 100644
--- a/usr/src/cmd/svc/svccfg/Makefile
+++ b/usr/src/cmd/svc/svccfg/Makefile
@@ -21,6 +21,7 @@
 
 #
 # Copyright (c) 2004, 2010, Oracle and/or its affiliates. All rights reserved.
+# Copyright 2017 Joyent, Inc.
 #
 
 MYPROG =	svccfg
@@ -134,6 +135,8 @@ $(PROG): $(OBJS) $(MAPFILES)
 	$(LINK.c) -o $@ $(OBJS) $(LDLIBS)
 	$(POST_PROCESS)
 
+$(POFILES): svccfg_grammar.h
+
 $(POFILE): $(POFILES)
 	cat $(POFILES) > $(POFILE)
 
-- 
2.21.0

